public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -160,7 +163,20 @@ export const filterRules = (
   }
 
   // If a namespace and group have rules that match the rules filters then keep them.
-  return filteredNamespaces.reduce<CombinedRuleNamespace[]>(reduceNamespaces(filterState), []);
+  const filteredRuleNamespaces: CombinedRuleNamespace[] = [];
+
+  try {
+    const matches = filteredNamespaces.reduce<CombinedRuleNamespace[]>(reduceNamespaces(filterState), []);
+    matches.forEach((match) => {
+      filteredRuleNamespaces.push(match);
+    });
+  } catch {
+    logError(SEARCH_FAILED_ERR, {
+      search: JSON.stringify(filterState),
+    });
+  }
+
+  return filteredRuleNamespaces;
```

Comment:
If I understand correctly, if the search fails we'll display an empty list of rules. I do wonder if we should display all rules instead? or empty and error?

Reply:
My reasoning was that it's much more likely to seem like the search _is_ working but just malformed. If we were to show the entire list then perhaps the user might interpret that as the search input not working at all.

Reply:
good point. Empty list will be a better incentive to change the search term. However, the user might still be convinced the search is not working as intended